main.go

```diff
@@ -133,13 +167,12 @@ func main() {
 		}
 	} else {
```

Comment:
refactor this please to:
if sanbox or watchnamespace == "", the cache is empty = cluster scoped
other wise, cache with watchnamespace. No need for this if/else block